src/components/EmojiSuggestions/index.js

```diff
@@ -0,0 +1,178 @@
+
+const propTypes = {
+    /** The index of the highlighted emoji */
+    highlightedEmoji: PropTypes.number,
+
+    /** Array of suggested emoji */
+    // eslint-disable-next-line react/forbid-prop-types
+    emojis: PropTypes.arrayOf(PropTypes.object).isRequired,
+
+    /** Callback called when the emoji is pressed */
+    onClose: PropTypes.func.isRequired,
+
+    /** Fired when the user selects an emoji */
+    onSelect: PropTypes.func.isRequired,
+
+    /** Emoji prefix that follows the colon  */
+    prefix: PropTypes.string.isRequired,
+
+    /** Show that we can use large emoji picker  */
+    isEmojiPickerLarge: PropTypes.bool.isRequired,
+
+    /** Show that we can use large emoji picker  */
+    shouldIncludeReportRecipientLocalTimeHeight: PropTypes.bool.isRequired,
+};
+
+const defaultProps = {
+    highlightedEmoji: 0,
+};
+const ROW_HEIGHT = 36;
+
+/**
+ * Select the correct color for text.
+ * @param {Boolean} isColored
+ * @returns {String | null}
+ */
+const colorOfText = isColored => ({backgroundColor: isColored ? colors.yellow : null});
+
+/**
+ * Select name as key.
+ * @param {Object} item
+ * @returns {String}
+ */
+const keyExtractor = item => item.name;
+
+/**
+ * @param {Number} numRows
+ * @param {Boolean} isEmojiPickerLarge
+ * @returns {Number}
+ */
+const measureHeightOfEmojiRows = (numRows, isEmojiPickerLarge) => {
+    if (isEmojiPickerLarge) {
+        return numRows * ROW_HEIGHT;
+    }
+    if (numRows > 2) {
+        // on small screens, we display a scrollable window with a height of 2.5 items, indicating that there are more items available beyond what is currently visible
+        return 2.5 * ROW_HEIGHT;
+    }
+    return numRows * ROW_HEIGHT;
+};
+
+/**
+ * Render a suggestion menu item component.
+ * @param {String} name
+ * @param {String} prefix
+ * @returns {Array}
+ */
+const getStyledTextsArray = (name, prefix) => {
+    const texts = [];
+    const prefixLocation = name.search(prefix);
+
+    if (prefixLocation === 0 && prefix.length === name.length) {
+        texts.push({text: prefix, isColored: true});
+    } else if (prefixLocation === 0 && prefix.length !== name.length) {
+        texts.push(
+            {text: name.slice(0, prefix.length), isColored: true},
+            {text: name.slice(prefix.length), isColored: false},
+        );
+    } else if (prefixLocation > 0 && prefix.length !== name.length) {
+        texts.push(
+            {text: name.slice(0, prefixLocation), isColored: false},
+            {
+                text: name.slice(prefixLocation, prefixLocation + prefix.length),
+                isColored: true,
+            },
+            {
+                text: name.slice(prefixLocation + prefix.length),
+                isColored: false,
+            },
+        );
+    } else {
+        texts.push({text: name, isColored: false});
+    }
+    return texts;
+};
+
+const EmojiSuggestions = (props) => {
+    /**
+     * Render a suggestion menu item component.
```

Comment:
I am not sure if js doc is relevant here? Could we just name `renderSuggestionMenuItem` ?